src/epoch.rs

```diff
@@ -263,27 +272,109 @@ impl Epoch {
+    }
+
+    #[must_use]
+    /// Converts self to another time scale
+    pub fn to_time_scale(&self, ts: TimeScale) -> Self {
+        match ts {
+            TimeScale::TAI => {
+                let mut new_epoch = self.clone();
+                // if previous time scale supported leap seconds: remove them
+                if self.time_scale.uses_leap_seconds() {
+                    new_epoch -= new_epoch.leap_seconds(true).unwrap_or(0.0) * Unit::Second;
+                }
+                // conversion to TAI: remove time scale reference point
+                new_epoch.duration -= ts.tai_reference_epoch().duration;
+                new_epoch.with_time_scale(TimeScale::TAI)
+            }
+            TimeScale::ET => {
+                // first convert back to TAI
+                let mut tai_epoch = self.to_time_scale(TimeScale::TAI);
+                // seconds past J2000
+                //TODO: this operation is not feasible is Self is not PAST J2000
+                let duration_since_j2000 = tai_epoch - J2000_REF_EPOCH;
+                let mut seconds_since_j2000 = duration_since_j2000.to_seconds();
+                for _ in 0..5 {
+                    seconds_since_j2000 += -NAIF_K
+                        * (NAIF_M0
+                            + NAIF_M1 * seconds_since_j2000
+                            + NAIF_EB * (NAIF_M0 + NAIF_M1 * seconds_since_j2000).sin())
+                        .sin();
+                }
+                // At this point, we have a good estimate of the number of seconds
+                // of this epoch. Reverse the algorithm:
+                let delta_et_tai = Self::delta_et_tai(
+                    seconds_since_j2000 - (TT_OFFSET_MS * Unit::Millisecond).to_seconds(),
+                );
+                // Match the SPICE by changing the UTC definition
+                Self {
+                    duration: (tai_epoch.duration.to_seconds() - delta_et_tai) * Unit::Second
+                        + J2000_TO_J1900_DURATION,
+                    time_scale: TimeScale::ET,
+                }
+            }
+            TimeScale::TDB => {
+                // first convert back to TAI
+                let mut tai_epoch = self.to_time_scale(TimeScale::TAI);
+                // seconds past J2000
+                //TODO: this operation is not feasible is Self is not PAST J2000
+                let duration_since_j2000 = tai_epoch - J2000_REF_EPOCH;
+                let seconds_since_j2000 = duration_since_j2000.to_seconds();
+                let gamma = Self::inner_g(seconds_since_j2000);
+                let delta_tdb_tai = gamma * Unit::Second + TT_OFFSET_MS * Unit::Millisecond;
+                tai_epoch += delta_tdb_tai;
+                tai_epoch += J2000_TO_J1900_DURATION; // TDB time scale is expressed past J2000
+                tai_epoch.with_time_scale(TimeScale::TDB)
+            }
+            ts => {
+                // first convert back to TAI
```

Comment:
Yes, I think that's correct. If I understand this correctly, it means that only ET and TDB have a different "tick" rate then the other time scales, i.e. one second in {TAI, UTC, GPS, GST, TT, ...} is one second in all of the other time scales, apart for TDB and ET. I think that is correct.